### PR TITLE
Core: handle intro exp + revert RI unknown trait handling + finish resolutions for prop 141

### PR DESCRIPTION
**Pull request by anonimal**
---
**By submitting this pull-request, I confirm the following:**

- I have read and understood the developer guide in [kovri-docs](https://github.com/monero-project/kovri-docs).
- I have checked that another pull-request for this purpose does not exist.
- I have considered and confirmed that this submission will be valuable to others.
- I accept that this submission may not be used and that this pull-request may be closed by the will of the maintainer.
- I give this submission freely under the BSD 3-clause license.
---

### CODE REVIEW

File: src/core/router/info.h [hunk 1]
```diff
@@ -241,6 +242,9 @@ struct RouterInfoTraits
         case Trait::IntroKey:
           return "ikey";
 
+        case Trait::IntroExp:
+          return "iexp";
```
Comment (coneiric) [inline src/core/router/info.h:246]: Thought about implementing as well, but in the spec for proposal 133, the key is `iexp{0,1,2}`. This won't match, will it?

The only other way I saw with our current implementation was a somewhat nasty `Trait::IntroExp0: return "iexp0";` and so on... It seemed better to ignore.

File: src/core/router/transports/impl.cc [hunk 2]
```diff
@@ -427,50 +425,6 @@ void Transports::HandleRequestComplete(
   }
 }
 
-void Transports::NTCPResolve(
```
Comment (coneiric) [inline src/core/router/transports/impl.cc:430]: I completely missed this, thank you for catching.

File: src/core/router/transports/impl.cc [hunk 3]
```diff
@@ -427,50 +425,6 @@ void Transports::HandleRequestComplete(
...
-      std::bind(
-          &Transports::HandleNTCPResolve,
-          this,
-          std::placeholders::_1,
-          std::placeholders::_2,
-          ident,
-          resolver));
-}
-
-void Transports::HandleNTCPResolve(
```
Comment (coneiric) [inline src/core/router/transports/impl.cc:448]: I completely missed this, thank you for catching.

File: src/core/router/info.h (hunk 1 above)
Comment (anonimal) [reply src/core/router/info.h:246]: > This won't match, will it?

It will. Look at lines 340-347 in router/info.cc

Comment (anonimal) [reply src/core/router/info.h:246]: FYI, it's not literally `iexp{0,1,2}`.

**Review by coneiric — DISMISSED** [review]
Just had a question about the introducer expiration impl. 

Thank you for revisions, and catching my hasty mistakes.

File: src/core/router/info.h (hunk 1 above)
Comment (coneiric) [reply src/core/router/info.h:246]: I know, was just using that for shorthand. I'll check the lines in router/info.cc

**Comment by anonimal** [comment]
Did you mean to request changes or approve?

**Comment by coneiric** [comment]
> Did you mean to request changes or approve?

Requested changes before seeing your replies. Mean to approve.

**Review by coneiric — APPROVED** [review]

**Pull request merged**